experiments/mine_dataflow.py

```
@@ -0,0 +1,230 @@
+                        know[d] = dict(slot_know[off])
+            # calls/ecalli clobber everything volatile — conservative: drop all
+            if i.op in ("Ecalli", "LoadImmJump", "LoadImmJumpInd", "JumpInd"):
+                know.clear()
+                slot_know.clear()
+                slot_in_reg.clear()
+                reg_holds.clear()
+            idx += 1
+
+    # ---------- offset relaxation estimate (fixpoint) ----------
+    # instructions with a 4-byte offset field at the end
+    def has_fixed_off(i):
```

Ruff (0.15.15)

[warning] Missing return type annotation for private function `has_fixed_off`

(ANN202)